src/IO/S3/PocoHTTPClient.cpp

```diff
@@ -254,9 +258,26 @@ void PocoHTTPClient::addMetric(const Aws::Http::HttpRequest & request, S3MetricT
 void PocoHTTPClient::makeRequestInternal(
     Aws::Http::HttpRequest & request,
     std::shared_ptr<PocoHTTPResponse> & response,
+    Aws::Utils::RateLimits::RateLimiterInterface * readLimiter ,
+    Aws::Utils::RateLimits::RateLimiterInterface * writeLimiter) const
+{
+    const auto request_configuration = per_request_configuration(request);
+    if (http_connection_pool_size && request_configuration.proxy_host.empty())
```

Comment:
did not understand why second condition is needed, add a comment?

Reply:
done